apps/shared/models/base.ts

```diff
@@ -31,6 +36,21 @@ export abstract class BaseModel<T> {
   } {
     throw Error('This was not implemented');
   }
+
+  /** Set the decoratedFor value on the prototype of the class. */
+  static decoratedFor(decoratedFor: symbol) {
+    this.prototype.decoratedForFunc(this, decoratedFor);
```

Comment:
Would it be better to just assign this as a static property, and not on a per-instance basis through the prototype? That would simplify the code. I can't think of a benefit with using prototype

Reply:
We want to have it as a class property so that we can check it during construction to make sure that some decorator has been set.

However, now that I looked closer at it, we don't need to leverage a prototype method to set the value on the prototype, we can just do it directly from the static method.

Reply:
I see what you are saying. It's because we can't know the class constructor from the derived classes in the Model base class